app/services/chf/create_derivatives_on_s3_service.rb

```diff
@@ -301,45 +301,78 @@ def create_image_jpg_derivative_future(width:, type_key:)
       end
     end
 
-
-    # Takes a pdf, makes a thumb. Includes small border around thumb, since this
-    # works better for typical white-bg print-like PDFs.
-    def create_pdf_jpg_thumb_future(width:, type_key:)
+    def create_pdf_thumb_future(width:, type_key:)
       defn = get_type_defn(type_key)
       style = defn.style
+
       output_path = Pathname.new(working_dir).join(type_key.to_s).sub_ext(defn.suffix).to_s
+      tmp_output_path = output_path.sub('.jpg', "_tmp.jpg")
+
```

Comment:
This adds a `tmp_output_path` because of the fact that you're now running multiple vips commands?

We need to make sure any temporary files are cleaned up and don't remain on disk after thumb is created. Otherwise bulk creation of thumbs can fill up disk space. This actually applies to the "final" thumb (on local disk) too. I haven't totally wrapped my head around what's going on or how this was handled in previous code. Using the ruby stdlib Tempfile class may be helpful.

Reply:
Actually as long as you are creating in `working_dir`, I think the surrounding code should take care of that, as the whole dir was created using ruby temp directory stuff. But maybe try to verify this is true?

Reply:
Sounds reasonable. I feel like we were deleting the temp folder containing all our working files, but it's definitely worth triple-checking.

Also I'm not sure I really need tmp_output_path. I feel like vips can run these edits in place and save the edited image to the same file path. Will take a quick look.